jscomp/ml/typedtree.mli

```diff
@@ -119,9 +119,8 @@ and expression =
 and exp_extra =
   | Texp_constraint of core_type
         (** E : T *)
-  | Texp_coerce of core_type option * core_type
-        (** E :> T           [Texp_coerce (None, T)]
-            E : T0 :> T      [Texp_coerce (Some T0, T)]
+  | Texp_coerce of core_type
```

Comment:
> Will this be problematic for the editor tooling/other things that read the compiler artifacts?

@zth good question.
This should be `Texp_coerce of unit * core_type` so for old compiler versions will produce a `.cmt` that has the rhs `core_type` in the right place (and the lhs apparing to be None).

Reply:
That'll make it work I take it?

Reply:
I have no idea how things are represented in OCaml and what can break.

Reply:
At runtime `unit` is `0` just like `None`, so the new representation will generate files that appear as valid also in the old representation.